src/main/java/org/datadog/jmxfetch/App.java

```diff
@@ -225,13 +232,17 @@ void start() {
             }
 
             // any SD configs waiting in pipe?
+            if(sdPipe == null && appConfig.getSDEnabled()) {
+                // If SD is enabled and the pipe is not open, retry opening pipe
+                sdPipe = newSdPipe();
+            }
             try {
```

Comment:
Little nit, but could do
int len = 0
if(sdPipe != null && ((len = sdPipe.available()) > 0)) {
   byte[] buffer....

No need to call sdPipe.available twice in succession.